src/napi/module.cpp

```diff
@@ -67,8 +76,16 @@ static napi_value setAPIVersionImpl(napi_env env, napi_callback_info info) {
   NAPI_OK_OR_RETURN_NULL(env, napi_get_value_int32(env, args[0], &apiVersion));
   int32_t headerVersion;
   NAPI_OK_OR_RETURN_NULL(env, napi_get_value_int32(env, args[1], &headerVersion));
-
-  FDB_OK_OR_RETURN_NULL(env, fdb_select_api_version_impl(apiVersion, headerVersion));
+  
+  if (previousApiVersion != 0) {
+    if (apiVersion != previousApiVersion) {
+      napi_throw_error(env, NULL, "foundationdb already initialized with another API version");
```

Comment:
... ditto here. If the versions don't match just let it call `fdb_select_api_version_impl` and let that throw.

Reply:
Semantics are different. fdb library doesn't allow you to call two times at all while we are allowing to call it several times with same apiVersion.

Reply:
Yes. I'm trying to take advantage of that so we return a canonical FDB error to the user if they try and set the version twice with different version numbers.

Alternately use `wrap_fdb_error(env, code, err*)` from util.h to make the error with `api_version_already_set` as the error code (a constant from foundationdb).